package/src/components/AutoCompleteInput/AutoCompleteSuggestionItem.tsx

```diff
@@ -0,0 +1,147 @@
+};
+
+export type EmojiItemType = {
+  name: string;
+  unicode: string;
+};
+
+export type AutoCompleteSuggestionItemPropsWithContext<
+  Us extends DefaultUserType = DefaultUserType,
+> = {
+  itemProps: CommandItemType | EmojiItemType | SuggestionUser<Us>;
+  type?: string;
+};
+
+const styles = StyleSheet.create({
+  args: {
+    fontSize: 14,
+  },
+  column: {
+    flex: 1,
+    justifyContent: 'space-evenly',
+    paddingLeft: 8,
+  },
+  container: {
+    alignItems: 'center',
+    flexDirection: 'row',
+    paddingHorizontal: 16,
+    paddingVertical: 8,
+  },
+  name: {
+    fontSize: 14,
+    fontWeight: 'bold',
+    paddingBottom: 2,
+  },
+  tag: {
+    fontSize: 12,
+    fontWeight: '600',
+  },
+  text: {
+    fontSize: 14,
+  },
+  title: {
+    fontSize: 14,
+    fontWeight: 'bold',
+    paddingRight: 8,
+  },
+});
+
+const AutoCompleteSuggestionItemWithContext = <Us extends DefaultUserType = DefaultUserType>({
+  itemProps,
+  type,
+}: AutoCompleteSuggestionItemPropsWithContext<Us>) => {
+  const {
+    theme: {
+      colors: { accent_blue, black, grey },
+      messageInput: {
+        suggestions: {
+          command: { args: argsStyle, container: commandContainer, title },
+          emoji: { container: emojiContainer, text },
+          mention: { avatarSize, column, container: mentionContainer, name: nameStyle },
+        },
+      },
+    },
+  } = useTheme();
+
+  if (type === 'mention') {
+    const { id, image, name, online } = itemProps as SuggestionUser<Us>;
+    return (
+      <View style={[styles.container, mentionContainer]}>
+        <Avatar image={image} name={name} online={online} size={avatarSize} />
+        <View style={[styles.column, column]}>
+          <Text style={[styles.name, { color: black }, nameStyle]} testID='mentions-item-name'>
+            {name || id}
+          </Text>
+        </View>
+        <AtMentions pathFill={accent_blue} />
+      </View>
+    );
+  } else if (type === 'emoji') {
+    const { name, unicode } = itemProps as EmojiItemType;
+    return (
+      <View style={[styles.container, emojiContainer]}>
+        <Text style={[styles.text, { color: black }, text]} testID='emojis-item-unicode'>
+          {unicode}
+        </Text>
+        <Text style={[styles.text, { color: black }, text]} testID='emojis-item-name'>
+          {` ${name}`}
+        </Text>
+      </View>
+    );
+  } else if (type === 'command') {
+    const { args, name } = itemProps as CommandItemType;
+    return (
+      <View style={[styles.container, commandContainer]}>
+        <AutoCompleteSuggestionCommandIcon name={name} />
+        <Text style={[styles.title, { color: black }, title]} testID='commands-item-title'>
+          {(name || '').replace(/^\w/, (char) => char.toUpperCase())}
+        </Text>
+        <Text style={[styles.args, { color: grey }, argsStyle]} testID='commands-item-args'>
+          {`/${name} ${args}`}
+        </Text>
+      </View>
+    );
+  } else {
+    return null;
+  }
```

Comment:
This should go as a separate task, since it will involve adding customization support for those individual components as well. I think its fine for now, its still quite readable :)

Reply:
But definitely worth doing in next iteration.

Reply:
I don't see the reason not to do it here, when it's being introduced as well. Customization for sure could be a separate task, but we already have so many massive components in the code base that I think it's worth making the effort to give things some room when we can. 

I'm open to being the odd one out here though, so you guys can decide what you'd like. But I don't think personally this should be merged as-is.

Reply:
![You_Shall_Not_Pass!_0-1_screenshot](https://user-images.githubusercontent.com/11586388/142029538-b929df33-fe46-461c-85ab-1dcba9e22d4e.jpeg)